source/llassetgen-rendering/main.cpp

```diff
@@ -0,0 +1,383 @@
+#include <algorithm>
+#include <iostream>
+#include <string>
+
+#include <ft2build.h>  // NOLINT include order required by freetype
+#include FT_FREETYPE_H
+
+#include <QApplication>
+#include <QBoxLayout>
+#include <QFormLayout>
+#include <QGroupBox>
+#include <QLabel>
+#include <QLineEdit>
+#include <QMainWindow>
+#include <QResizeEvent>
+#include <QValidator>
+
+#include <glm/glm.hpp>
+
+#include <glbinding/ContextInfo.h>
+#include <glbinding/Version.h>
+#include <glbinding/gl/gl.h>
+
+#include <globjects/base/File.h>
+#include <globjects/globjects.h>
+#include <globjects/logging.h>
+
+#include <globjects/Buffer.h>
+#include <globjects/Program.h>
+#include <globjects/Shader.h>
+#include <globjects/Texture.h>
+#include <globjects/VertexArray.h>
+#include <globjects/VertexAttributeBinding.h>
+
+#include <llassetgen/llassetgen.h>
+
+#include "WindowQt.h"
+
+using namespace gl;
+
+/* Taken from cginternals/globjects,
+ * example: qtexample, texture
+ * edited
+ */
+
+/*
+ * This class is used to display GL rendered content in a Qt Window.
+ * It overrides all functions from WindowQt which use GL functionality.
+ * It handles the GL context and loads the shader from file.
+ * Some functions are connected to the GUI using Qt signals and slots.
+ */
+class Window : public WindowQt {
+   public:
+    explicit Window(const QSurfaceFormat &format) : WindowQt(format) {
+        m_samplerIndex = 0;
+        m_texture = nullptr;
+        m_backgroundColor = glm::vec4(1.f, 1.f, 1.f, 1.f);
+        m_fontColor = glm::vec4(0.f, 0.f, 0.f, 1.f);
+    }
+
+    virtual ~Window() {}
+
+    virtual void initializeGL() override {
+        globjects::init();
+
+        std::cout << std::endl
+                  << "OpenGL Version:  " << glbinding::ContextInfo::version() << std::endl
+                  << "OpenGL Vendor:   " << glbinding::ContextInfo::vendor() << std::endl
+                  << "OpenGL Renderer: " << glbinding::ContextInfo::renderer() << std::endl
+                  << std::endl;
+
+        globjects::DebugMessage::enable();
+
+#ifdef __APPLE__
+        globjects::Shader::clearGlobalReplacements();
+        globjects::Shader::globalReplace("#version 140", "#version 150");
+
+        globjects::debug() << "Using global OS X shader replacement '#version 140' -> '#version 150'" << std::endl;
+#endif
+
+        // get glyph atlas
+
+        // TODO load using own png loader instead of Qt (blocked: wait for this feature to be merged into master, then
+        // pull). TODO: Make sure this file structure is the same on other OS
+        auto path = QApplication::applicationDirPath();
+        auto *image = new QImage(path + "/../../data/llassetgen-rendering/testfontatlas_rgb.png");
```

Comment:
If we want to include a default font atlas long-term, we should probably include it in the executable or figure out how to install into and reference the standard directories (stuff like this would probably go into `/usr/[local/]share` on Linux and macOS).

Reply:
This path finds the executable path and works relative from there.
It's not about including a default atlas long-term, but asking the user to specify a path where to load from. Until then, we just load a png as a mock. That's the plan :D